students/alexLaws/lesson04/mailroom.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+donor_list = []
+donor_list.append({'Name': 'Ryan Moore', 'Donation': 500})
+donor_list.append({'Name': 'Ryan Moore', 'Donation': 250})
+donor_list.append({'Name': 'Ted Laws', 'Donation': 1000})
+donor_list.append({'Name': 'Ted Laws', 'Donation': 100})
+donor_list.append({'Name': 'Ben Snell', 'Donation': 150})
+donor_list.append({'Name': 'Andrew Crawford', 'Donation': 2000})
+donor_list.append({'Name': 'Andrew Crawford', 'Donation': 2000})
+donor_list.append({'Name': 'Andrew Crawford', 'Donation': 4000})
+donor_list.append({'Name': 'Beth Ross', 'Donation': 400})
```

Comment:
I'm going to review below code as is.. but I would really recommend to reconsider the data structure chosen here which will change a lot of your code.

The idea to use dict for donor DB is to make things much easier and because you are still dealing with a list it is not easier.

here are two options: 
```
donors = {
    "Ryan Moore": [234, 333, 234343],
    "Ted Laws": [444, 100, 300]
}

donors = {
    "Ryan Moore": {"donations": [234, 333, 234343]},
    "Ted Laws": {"donations": [444, 100, 300]}
}
```

first option is bare minimum for your needs and simple, second is an extendable version in case you would like to associate more data to your donor.

notice that donor name is a natural key, look up by key is very cheap and you are not duplicating data

once you go with a `dict` for your data structure, next check out `defaultdict` and how it can help you here 🙂